lib/msal-browser/src/cache/BrowserStorage.ts

```diff
@@ -138,13 +135,48 @@ export class BrowserStorage implements ICacheStorage {
      * Will retrieve frm cookies if storeAuthStateInCookie is set to true.
      * @param key
      */
-    getItem(key: string): string {
-        const msalKey = this.generateCacheKey(key);
-        const itemCookie = this.getItemCookie(msalKey);
-        if (this.cacheConfig.storeAuthStateInCookie && itemCookie) {
-            return itemCookie;
+    getItem(key: string, type: string): string | object {
```

Comment:
Per discussion with @pkanher617, instead of baking this business logic into these functions, you should instead create an abstract class with functions for each of the `case` statements, and then have `BrowserStorage` implement `ICacheStorage` and extend the abstract class.

Reply:
That way consumers can use a use-case specific function instead of passing a magic string, and that way if applications provide their own storage implementations, they don't have to know about this logic.

Reply:
We will revisit this when we add the feature for consumer driven storage.

Reply:
We already have consumer driven storage (the extensions library).

Reply:
@sameerag @pkanher617 Let's talk about this next design meeting.

Reply:
We also have other storage scenarios for the web such as VS code extensions that we will want to support, which would fall under "user-provided storage"

Reply:
The abstract class is not implemented in the follow up PR #1771. Regarding abstracting `getItem` implementation, I agree we can improve it for `custom storage`, however I would like to track it separately and as an enhancement.

Reply:
We can prioritize if we can arrive at a design soon but considering the timelines to get a working branch earlier, I would like to close this PR.